contrib/correlation-context/correlationcontext/__init__.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .correlation_context import CorrelationContext
```

Comment:
@c24t do you think this name makes sense, or we want `opencensus.common` as the namespace?
I didn't put it under trace or tags, since correlationcontext could be used by both of them.
I probably need to move `opencensus.trace.tracestate` to a common place as well.

Reply:
I think it makes sense to put this in common instead of a new top-level namespace, but I don't understand how `CorrelationContext` is going to be used in the library yet, so up to you. If not for tags, would you expect this to go in the `trace.propagation` namespace? If so, does it make sense to have a `common.propagation` namespace?

I'm surprised to see you put context correlation in a separate distribution since trace context is part of the core library. Would you expect to move trace context out too, or are they different in a way that I don't understand?

Reply:
This should probably also be a full path import to be consistent with other imports unless there's a reason you need this for the namespace package.

Reply:
Changed to full path import.

Reply:
Adding @SergeyKanzhelev for discussion.

Here goes my thinking:
1. CorrelationContext is a generic context that both OpenCensus and end user could consume.
2. Put tracecontext and correlationcontext in separate packages (under their own namespace), so we can promote them from https://github.com/w3c/trace-context/#reference-implementations and encourage people to reuse the libraries as much as possible.
3. The namespace common.propagation should contain the logic built on top of tracecontext/correlationcontext to handle protocol specific logic (e.g. gRPC propagation for CorrelationContext, which could be used for tags propagation and customer property progation).

Reply:
@c24t Python3.7 has introduced a [contextlib](https://docs.python.org/3/library/contextlib.html), which I encountered while exploring asyncio support for Census context propagation.
Probably we would want to put tracecontext and correlationcontext into contextlib in the long run?

Reply:
I've got some work to do to understand correlation context, in particular: how it interacts with the trace context, how it compares to the context API we expose in other clients (which is [grpc's `Context`](https://grpc.io/grpc-java/javadoc/io/grpc/Context.html) in java), and how users would use it outside of OC. I'm willing to believe it should be a top level package, but we might have to revisit the package structure when we add it to other clients.

> Probably we would want to put tracecontext and correlationcontext into contextlib in the long run?

As in add it to the standard library? This doesn't seem like a good fit since `contextlib` provides tools for writing context managers, and isn't a collection of runtime contexts (although I'm surprised to see `redirect_stdout` in the library). We could write a context manager for working with opencensus contexts now, even without `contextlib`, and include it in this library.

Reply:
These are valid points, thanks you @c24t.
For now, we can move correlation context to opencensus.common namespace, and continue the package refactor work. Does this sound okay to you?

Reply:
That sounds good, and either namespace works for me.

Reply:
Great. Moved to opencensus.common.correlationcontext namespace.